dvc/scm/git/backend/dulwich/asyncssh_vendor.py

```diff
@@ -48,6 +48,45 @@ def close(self):
         self.conn.close()
 
 
+# NOTE: Github's SSH server does not strictly comply with the SSH protocol.
+# When validating a public key using the rsa-sha2-256 or rsa-sha2-512
+# signature algorithms, RFC4252 + RFC8332 state that the server should respond
+# with the same algorithm in SSH_MSG_USERAUTH_PK_OK. Github's server always
+# returns "ssh-rsa" rather than the correct sha2 algorithm name (likely for
+# backwards compatibility with old SSH client reasons). This behavior causes
+# asyncssh to fail with a key-mismatch error (since asyncssh expects the server
+# to behave properly).
+#
+# See also:
+#   https://www.ietf.org/rfc/rfc4252.txt
+#   https://www.ietf.org/rfc/rfc8332.txt
```

Comment:
Should note that this appears to be specific to github. gitlab's SSH server returns the proper algorithm names in the auth responses.

This is also one of those cases where the RFC's do also note that some existing SSH implementations in the wild may just always use "ssh-rsa", and that it's up to individual client/server implementations to decide whether or not to allow or reject those requests & responses.